src/sail_json_backend/json.ml

```diff
@@ -232,12 +232,55 @@ let parse_encdec i mc format =
       end
   | _ -> assert false
   
+let extract_operands k =
+  let components = Hashtbl.find assembly k in
+  let regex = Str.regexp ".+(\\(.*\\))" in
+  let rec pre_filter acc = function
+    | [] -> List.rev acc
+    | hd :: tl ->
+      if String.trim hd = "spc" then
+        tl
+      else
+        pre_filter ( hd :: acc ) tl
+  in
+  let filtered_components = pre_filter [] components in 
+  let rec extract acc = function
+    | [] -> List.rev acc
+    | hd :: tl ->
+      if Str.string_match regex hd 0 then
+        let operand = Str.matched_group 1 hd in
+        let trimmed_operand = 
+          try
+            let comma_index = String.index operand ',' in
+            let trimmed = String.sub operand 0 comma_index in
+            debug_print ("Trimmed operand using comma: " ^ trimmed);
+            trimmed
+          with Not_found -> 
+          debug_print ("No comma found in operand: " ^ operand);
+          operand
```

Comment:
I'd prefer that these lines be indented to align with the statements in the `try` block.